src/components/MultiHopTrade/components/TradeInput/components/QuoteTimer.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+
+import { CircularProgress } from '@/components/CircularProgress/CircularProgress'
+import {
+  TRADE_QUOTE_REFRESH_INTERVAL_MS,
+  TRADE_QUOTE_TIMER_UPDATE_MS,
+} from '@/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeRates'
+import { selectLastRefreshTime } from '@/state/slices/tradeQuoteSlice/selectors'
+import { useAppSelector } from '@/state/store'
+
+type QuoteTimerProps = {
+  size?: string | number
+}
+
+const getElapsed = (lastRefreshTime: number) => {
+  const elapsed = Date.now() - lastRefreshTime
+  return Math.max(0, TRADE_QUOTE_REFRESH_INTERVAL_MS - elapsed)
+}
+
+export const QuoteTimer = ({ size = '6' }: QuoteTimerProps) => {
+  const lastRefreshTime = useAppSelector(selectLastRefreshTime)
```

Comment:
And here, instead of relying on the store, we can get the `useQuery` data, to get the current polling time accordingly